warehouse/jobs/runner.go

```diff
@@ -35,75 +36,52 @@ func InitWarehouseJobsAPI(
+	a.maxCleanUpRetries = config.GetInt("Warehouse.jobs.maxCleanUpRetries", 5)
+	a.maxQueryRetries = config.GetInt("Warehouse.jobs.maxQueryRetries", 3)
+	a.maxAttemptsPerJob = config.GetInt("Warehouse.jobs.maxAttemptsPerJob", 3)
+	a.retryTimeInterval = config.GetDuration("Warehouse.jobs.retryTimeInterval", 10, time.Second)
+	a.asyncJobTimeOut = config.GetDuration("Warehouse.jobs.asyncJobTimeOut", 300, time.Second)
 }
 
 func (a *AsyncJobWh) getTableNamesBy(sourceID, destinationID, jobRunID, taskRunID string) ([]string, error) {
 	a.logger.Infof("[WH-Jobs]: Extracting table names for the job run id %s", jobRunID)
 	var tableNames []string
 	var err error
-	query := `SELECT id
-		FROM
-		` + warehouseutils.WarehouseUploadsTable + `
-		WHERE metadata->>'source_job_run_id'=$1
-			AND metadata->>'source_task_run_id'=$2
-			AND metadata in (
-				SELECT
-					metadata
-				FROM
-				` + warehouseutils.WarehouseUploadsTable + `
-				WHERE source_id=$3
-				AND destination_id=$4
-			)`
+
+	query := `SELECT table_name FROM ` + warehouseutils.WarehouseTableUploadsTable + ` WHERE wh_upload_id IN ` +
+		` (SELECT id FROM ` + warehouseutils.WarehouseUploadsTable + ` WHERE metadata->>'source_job_run_id'=$1
+				AND metadata->>'source_task_run_id'=$2
+				AND source_id=$3
+				AND destination_id=$4)`
 	a.logger.Debugf("[WH-Jobs]: Query is %s\n", query)
-	rows, err := a.dbHandle.Query(query, jobRunID, taskRunID, sourceID, destinationID)
+	rows, err := a.dbHandle.QueryContext(a.context, query, jobRunID, taskRunID, sourceID, destinationID)
 	if err != nil {
-		a.logger.Errorf("[WH-Jobs]: Error carrying out the query %s ", query)
-		return nil, err
+		a.logger.Errorf("[WH-Jobs]: Error executing the query %s with error %v", query, err)
+		return tableNames, err
 	}
-	defer rows.Close()
+	defer func() { _ = rows.Close() }()
 	for rows.Next() {
-		var uploadId string
-		err := rows.Scan(&uploadId)
-		if err != nil {
-			a.logger.Errorf("[WH-Jobs]: Error carrying the scan operation to uploadId\n")
-			return nil, err
-		}
-		query = fmt.Sprintf(`select table_name from %s where wh_upload_id=$1`, warehouseutils.WarehouseTableUploadsTable)
-		tables, err := a.dbHandle.Query(query, uploadId)
+		var tableName string
+		err = rows.Scan(&tableName)
 		if err != nil {
-			a.logger.Errorf("[WH-Jobs]: Error carrying out the query %s ", query)
-			return nil, err
-		}
-		for tables.Next() {
-			var tableName string
-			err = tables.Scan(&tableName)
-			if err != nil {
-				a.logger.Errorf("[WH-Jobs]: Error carrying the scan operation to tablename\n")
-				return nil, err
-			}
-			if !contains(tableNames, tableName) {
-				tableNames = append(tableNames, tableName)
-			}
-
+			a.logger.Errorf("[WH-Jobs]: Error scanning the rows %s", err.Error())
+			return tableNames, err
 		}
+		tableNames = append(tableNames, tableName)
+	}
+	if err = rows.Err(); err != nil {
+		a.logger.Errorf("[WH-Jobs]: Error iterating the rows %s", err.Error())
+		return tableNames, err
```

Comment:
Can we return `nil` instead of `tableNames` in the scenarios where `err` is not `nil`?